ansible/roles/config/templates/config.toml.j2

```diff
@@ -1,57 +1,221 @@
+priv_validator_file = "config/priv_validator.json"
+
+# Path to the JSON file containing the private key to use for node authentication in the p2p protocol
+node_key_file = "config/node_key.json"
+
+# Mechanism to connect to the ABCI application: socket | grpc
+abci = "socket"
+
+# TCP or UNIX socket address for the profiling server to listen on
+prof_laddr = ""
+
+# If true, query the ABCI app on connecting to a new peer
+# so the app can decide if we should keep the connection or not
+filter_peers = false
+
+##### advanced configuration options #####
+
+##### rpc server configuration options #####
 [rpc]
+
+# TCP or UNIX socket address for the RPC server to listen on
 laddr = "tcp://0.0.0.0:46657"
 
+# TCP or UNIX socket address for the gRPC server to listen on
+# NOTE: This server only supports /broadcast_tx_commit
+grpc_laddr = ""
+
+# Activate unsafe RPC commands like /dial_seeds and /unsafe_flush_mempool
+unsafe = false
+
+##### peer to peer configuration options #####
+[p2p]
+
+# Address to listen for incoming connections
+laddr = "tcp://0.0.0.0:46656"
+
+# Comma separated list of seed nodes to connect to
+seeds = "{{ seeds | default() }}"
+
+# Comma separated list of nodes to keep persistent connections to
+{% set comma = joiner(",") %}persistent_peers = "{% for host in ((groups[testnet_name]|default([]))+(groups['tag_Environment_'~(testnet_name|regex_replace('-','_'))]|default([])))|difference(inventory_hostname) %}{{ comma() }}{{hostvars[host]["nodeid"]}}@{{hostvars[host]["inventory_hostname"]}}:46656{% endfor %}"
+
+# Path to address book
+addr_book_file = "config/addrbook.json"
+
+# Set true for strict address routability rules
+addr_book_strict = true
+
+# Time to wait before flushing messages out on the connection, in ms
+flush_throttle_timeout = 100
+
+# Maximum number of peers to connect to
+#max_num_peers = 50
+max_num_peers = 300
+
+# Maximum size of a message packet payload, in bytes
 {% if service == 'tendermint' %}
+max_msg_packet_payload_size = 65536
+{% else %}
+max_msg_packet_payload_size = 1024
+{% endif %}
+
+# Rate at which packets can be sent, in bytes/second
+{% if service == 'tendermint' %}
+send_rate = 51200000 # 50 MB/s
+{% else %}
+send_rate = 512000
+{% endif %}
+
+# Rate at which packets can be received, in bytes/second
+{% if service == 'tendermint' %}
+recv_rate = 51200000 # 50 MB/s
+{% else %}
+recv_rate = 512000
+{% endif %}
+
+# Set true to enable the peer-exchange reactor
+pex = true
+
+# Seed mode, in which node constantly crawls the network and looks for
+# peers. If another node asks it for addresses, it responds and disconnects.
+#
+# Does not work if the peer-exchange reactor is disabled.
+seed_mode = false
 
+##### mempool configuration options #####
 [mempool]
+
+{% if service == 'tendermint' %}
 recheck = false
+{% else %}
+recheck = true
+{% endif %}
+recheck_empty = true
 broadcast = true
+{% if service == 'tendermint' %}
```

Comment:
so if the service is tendermint, we set wal_dir to ""? why?

Reply:
Bucky told me so. ;) For performance testing.